libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -543,17 +542,54 @@ void CompilerUtils::convertType(
 		else if (targetTypeCategory == Type::Category::FixedPoint)
 		{
 			solAssert(
-				stackTypeCategory == Type::Category::Integer || 
+				stackTypeCategory == Type::Category::Integer ||
 				stackTypeCategory == Type::Category::RationalNumber ||
 				stackTypeCategory == Type::Category::FixedPoint,
 				"Invalid conversion to FixedMxNType requested."
 			);
-			//shift all integer bits onto the left side of the fixed type
+
 			FixedPointType const& targetFixedPointType = dynamic_cast<FixedPointType const&>(_targetType);
 			if (auto typeOnStack = dynamic_cast<IntegerType const*>(&_typeOnStack))
-				if (targetFixedPointType.numBits() > typeOnStack->numBits())
+            {
+				if (targetFixedPointType.numBits() > typeOnStack->numBits() + targetFixedPointType.fractionalDigits() * 3)
```

Comment:
Can you explain the calculation here? I think it is more complicated than just multiplying the fractional digits by 3.

Reply:
Yeah, the idea was to save a check for extra bits, but I just completely messed it up. I will remove it.